packages/vscode-extension/src/project/metro.ts

```diff
@@ -350,11 +351,21 @@ export class Metro implements Disposable {
     // be 1, reanimated worklet runtime would get 2, etc.)
     // The most recent runtimes are listed first, so we can pick the first one with title
     // that that starts with "React Native Bridge" (which is the main runtime)
-    for (const page of listJson) {
-      if (page.reactNative && page.title.startsWith("React Native Bridge")) {
-        return page.webSocketDebuggerUrl;
+    const newDebuggerPages = listJson.filter(
+      (page) => page.reactNative && page.title.startsWith("React Native Bridge")
+    );
+    if (newDebuggerPages.length > 0) {
+      // Expo go apps would report at least two pages, first one being the Expo Go host runtime
+      // If we detect Expo Go package (using description field), we want to pick the second page
+      // otherwise we pick the first one
+      const description = newDebuggerPages[0].description;
+      const isExpoGo = description === EXPO_GO_BUNDLE_ID || description === EXPO_GO_PACKAGE_NAME;
+      if (isExpoGo && newDebuggerPages.length > 1) {
+        return newDebuggerPages[1].webSocketDebuggerUrl;
```

Comment:
How do you know that the second runtime is the correct one? Could you provide source?

Reply:
There's no source and it is based on observation. What we observe is documented in the comment that expo go spins up two runtimes. The first one is host runtime and it seem logical that it will be created first